backend/test/users_test.js

```diff
@@ -296,3 +296,18 @@ describe("Tests add student user", function() {
       });
   });
 });
+
+describe("Test retrieve student data", function() {
+  it("responds with valid", function() {
+    return users.getStudentData(2).then(function(res) {
+      let found = false;
+      let searchingFor = {
+        major: [],
+        minor: [],
+        courses: [{ course_code: "ECSE 428", semester: "winter" }]
+      };
+      if (JSON.stringify(course) == JSON.stringify(searchingFor)) found = true;
```

Comment:
As it's structured, the test is likely going to break as soon as we start adding more stuff into the database

Reply:
Try inserting random values into the necessary tables and then retrieving those in your test. Like this for example: 

it('responds with valid', async function(){
        connection = await mysql.getNewConnection()
        connection.query(`INSERT INTO courses (course_code,title, department) VALUES 
        (?,?,?) ON DUPLICATE KEY UPDATE course_code=course_code;`,['ECSE 428','Software engineering in practice', 'ECSE']);
        connection.query(`INSERT INTO tags(name) VALUES
        (?);`,['Engineering'])
        connection.query(`INSERT INTO course_tag (course_code, tag_name) VALUES
        (?,?);`,['ECSE 428', 'Engineering'])
        return courses.queryCourseByTag('Engineering').then(function(res){
            let found = false;
            let searchingFor = {course_code: 'ECSE 428'};
            for (course in res){
                if (JSON.stringify(course) == JSON.stringify(searchingFor))
                    found = true;
            }
            assert(true,found)
        });
    });
Make sure you insert random data in it so that such values are not already in the database and that you remove that random data afterwards (In my case ECSE 428 is pretty much a constant entry that should always be in the database so I did not bother, but it would have been better practice to insert some random value eg randomcourse123456) 
Also I'm not sure but shouldn't you be comparing JSON.stringify(res) == JSON.stringify(searchingFor) instead of course? what is course? :O

Reply:
For easier reading of what he wrote @zerfuth 

```node
it("responds with valid", async function() {
  connection = await mysql.getNewConnection();
  connection.query(
    `INSERT INTO courses (course_code,title, department) VALUES (?,?,?) ON DUPLICATE KEY UPDATE course_code=course_code;`,
    ["ECSE 428", "Software engineering in practice", "ECSE"]
  );
  connection.query(`INSERT INTO tags(name) VALUES (?);`, ["Engineering"]);
  connection.query(
    `INSERT INTO course_tag (course_code, tag_name) VALUES (?,?);`,
    ["ECSE 428", "Engineering"]
  );
  return courses.queryCourseByTag("Engineering").then(function(res) {
    let found = false;
    let searchingFor = { course_code: "ECSE 428" };
    for (course in res) {
      if (JSON.stringify(course) == JSON.stringify(searchingFor)) found = true;
    }
    assert(true, found);
  });
});

```